### PR TITLE
Introduce responsive props + internal refactors of components

### PR DESCRIPTION
**Pull request by geoffreydhuyvetters**
### Added

- Added the possibility to pass an object as styling prop, to use for hover states and breakpoints

```js

backgroundColor={{
  hover: 'green',
  s: 'yellow',
  default: 'orange'
}}

```

### Changed

- did some internal component refactors while at it

the breakpoint with their values are defined in the theme

### CODE REVIEW

File: src/components/publiq-ui/Panel.js [hunk 1]
```diff
@@ -1,27 +1,61 @@
...
-    {...getStackProps(props)}
-    className={className}
-    marginBottom={4}
-  >
-    {children}
-  </Stack>
-);
+const Panel = ({ children, className, ...props }) => {
+  const parsedChildren =
+    Children.count(children) === 1 ? <>{children}</> : children;
```
Comment (simon-debruijn) [inline src/components/publiq-ui/Panel.js:10]: I see this line is added in several components. But why is this needed?

**Review by simon-debruijn — APPROVED** [review]

File: src/components/publiq-ui/Panel.js (hunk 1 above)
Comment (geoffreydhuyvetters) [reply src/components/publiq-ui/Panel.js:10]: It's a fix for passing a single element to Panel and ListItem (these are Inlines now).
